### PR TITLE
Modified the smart contract and tested with eth(deposit/withdraw) full cycle

### PR DESCRIPTION
**Pull request by KarishmaBothara**

### CODE REVIEW

File: scripts/ethEndToEndTest.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                    resolve();
+                });
+            }
+        });
+    });
+    /********************* Set up complete *******************************************/
+
+    /********** new Token Id that would be minted on deposition at CENNZnet **********/
+    const testTokenId = await api.query.genericAsset.nextAssetId();
```
Comment (holygits) [inline scripts/ethEndToEndTest.js:84]: could do query `api.query.erc20peg.erc20ToAssetId(ethAddress)` to check if it's mapped already

File: scripts/ethEndToEndTest.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    /***************** Make Deposit on CENNZnet **********************/
+    let eventClaimId;
+    let nonce = await api.rpc.system.accountNextIndex(alice.address);
+    await new Promise(async (resolve, reject) => {
+        await api.tx.erc20Peg.depositClaim(depositTxHash, claim).signAndSend(alice, {nonce}, async ({status, events}) => {
+            if (status.isInBlock) {
+                for (const {event: {method, section, data}} of events) {
+                    console.log('\t', `: ${section}.${method}`, data.toString());
+                    if (section === 'erc20Peg' && method == 'Erc20Claim') {
+                        eventClaimId = data[0];
```
Comment (holygits) [inline scripts/ethEndToEndTest.js:102]: is it possible to checker the claimer/sender address here also?

File: scripts/ethEndToEndTest.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                    if (event.section === 'ethBridge' && event.method === 'Verified') {
+                        unsubHeads();
+                        resolve();
+                    }
+                });
+            });
+        });
+    });
+    await new Promise(async (resolve, reject) => {
+        let nonce = await api.rpc.system.accountNextIndex(alice.address);
```
Comment (holygits) [inline scripts/ethEndToEndTest.js:138]: wondering why query nonce separately?

File: contracts/CENNZnetBridge.sol [hunk 4]
```diff
@@ -102,9 +102,10 @@ contract CENNZnetBridge is Ownable {
         validators[proof.validatorSetId] = newValidators;
         activeValidatorSetId = proof.validatorSetId;
 
-        // return any accumlated fees to the sender as a reward
+        // return any accumulated fees to the sender as a reward
         uint reward = address(this).balance;
-        payable(msg.sender).transfer(reward);
+        (bool sent, ) = msg.sender.call{value: reward}("");
```
Comment (holygits) [inline contracts/CENNZnetBridge.sol:107]: wondering if this will succeed when `reward` is `0`? (we would want it to)

Comment (KarishmaBothara) [reply contracts/CENNZnetBridge.sol:107]: Tested this.. on ropsten
https://ropsten.etherscan.io/tx/0x71891e2a496aee6f4f6d00092222b7802711ddfc6540bc8bf9ac7a37de3ed3b9

File: scripts/ethEndToEndTest.js (hunk 1 above)
Comment (KarishmaBothara) [reply scripts/ethEndToEndTest.js]: done

File: scripts/ethEndToEndTest.js (hunk 2 above)
Comment (KarishmaBothara) [reply scripts/ethEndToEndTest.js:103]: done

File: scripts/ethEndToEndTest.js (hunk 3 above)
Comment (KarishmaBothara) [reply scripts/ethEndToEndTest.js]: undone this

**Review by holygits — APPROVED** [review]

**Pull request merged**